sample/dynamic.go

```diff
@@ -39,6 +39,7 @@ func (d *DynamicSampler) Start() error {
 	d.dynsampler = &dynsampler.AvgSampleRate{
 		GoalSampleRate:         int(d.sampleRate),
 		ClearFrequencyDuration: time.Duration(d.clearFrequency),
+		MaxKeys:                d.Config.MaxKeys,
```

Comment:
Is this a separate bug fix?

Reply:
Kinda? I realized while doing this that some samplers didn't allow this field to be configured, and I wanted all the documentation to be consistent, so I did it along the way.